edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProvider.cs

```diff
@@ -31,11 +31,13 @@ public async Task<byte[]> GetLogs(string id, ModuleLogOptions logOptions, Cancel
                 logOptions.Filter.IncludeTimestamp = Option.Some(true);
             }
 
-            Stream logsStream = await this.runtimeInfoProvider.GetModuleLogs(id, false, logOptions.Filter.Tail, logOptions.Filter.Since, logOptions.Filter.Until, logOptions.Filter.IncludeTimestamp, cancellationToken);
-            Events.ReceivedStream(id);
+            using (Stream logsStream = await this.runtimeInfoProvider.GetModuleLogs(id, false, logOptions.Filter.Tail, logOptions.Filter.Since, logOptions.Filter.Until, logOptions.Filter.IncludeTimestamp, cancellationToken))
```

Comment:
Is there any convention on whether the `using` should be in the code that creates the `IDisposable` (here) or the code that "consumes" the `IDisposable` (`GetProcessedLogs`) ?

Reply:
If I got your question right, there is no convention. In our case we pass the "active" `Stream` up the stack, so we pass the responsibility of disposing the stream to the "consumer".